### PR TITLE
feat(tools): add dependency-free BM25 ranker

### PR DESCRIPTION
**Pull request by edelauna**
## Related GitHub Issue

Relates to #575

## Description

Adds a dependency-free, pure-JavaScript BM25 ranker as the always-available baseline for MCP tool search. This PR is limited to the foundational ranker described in issue #575; it does not implement downstream routing or threshold-gating work.

### Implementation

- Adds the `ToolDoc` type with `serverName`, `toolName`, and `description` fields.
- Adds the `Ranker` interface with `rank(query, items, k)` returning the top-ranked `ToolDoc` results.
- Implements standard BM25 scoring with `k1 = 1.5` and `b = 0.75`, with no external npm dependencies.
- Tokenizes the combined server name, tool name, and description by lowercasing and splitting on non-word characters.
- Builds the inverted index lazily on the first `rank()` call and rebuilds it only when the input array reference changes.
- Handles empty queries, empty item sets, no-match queries, zero or limited `k`, empty descriptions, and stable input ordering for equal scores.

## Tests and Validation

- Adds focused unit tests covering BM25 relevance ordering, edge cases, top-`k` truncation, stable tie ordering, case-insensitive punctuation-aware tokenization, empty descriptions, and reference-based index rebuilding.
- Validation: `cd src && npx vitest run services/tools/__tests__/Bm25Ranker.spec.ts`

## Summary by CodeRabbit

* **New Features**
  * Added relevance-based search for available tools.
  * Search results now account for tool names and descriptions, with case- and punctuation-insensitive matching.
  * Results are limited to the requested number and consistently ordered when relevance is equal.
  * Empty searches and searches without matches return no results.

* **Bug Fixes**
  * Improved result accuracy when tool information changes or descriptions are unavailable.

### CODE REVIEW

File: src/services/tools/Bm25Ranker.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			score += inverseDocumentFrequency * ((termFrequency * (this.k1 + 1)) / (termFrequency + normalization))
+		}
+
+		return score
+	}
+}
+
+function tokenize(value: string): string[] {
+	return value.toLowerCase().split(/\W+/).filter(Boolean)
+}
```
Comment (taltas) [inline src/services/tools/Bm25Ranker.ts:90]: Could we treat underscores as separators here so common MCP tool names such as `create_issue` match a query like `create issue`, and add a regression test?
